src/Orleans/RelationalStorage/OrleansRelationalExtensions.cs

```diff
@@ -497,8 +497,10 @@ public static class OrleansRelationalExtensions
             {
                 var ip = selector.GetValue<string>("Address");
                 var port = selector.GetValue<int>("ProxyPort");
+                var gen = selector.GetValue<int>("Generation");
```

Comment:
This ``GetValue`` parses results of the query from the wire protocol. You can't see the query here, but is retrieved from the query cache via ``OrleansRelationalConstants.GetConstant(storage.InvariantName, OrleansRelationalConstants.ActiveGatewaysQuery)`` to which it is loaded from the database.

You need add this ``Generation`` to the query definition in ``CreateOrleansTables_SqlServer.sql``. I'll add the new query here:
```
INSERT INTO [OrleansQuery]([Key], [Query], [Description])
VALUES
(
	'ActiveGatewaysQueryKey',
	N'SET NOCOUNT ON;
	SELECT
            [Address],
            [ProxyPort],
            [Generation]
      FROM
		[MembershipReadAll]
      WHERE
		[DeploymentId] = @deploymentId AND @deploymentId IS NOT NULL
        AND [Status]   = @status AND @status IS NOT NULL;',
	N''
);
```
Few added notes:
1) When one uses these new binaries (e.g. in 1.1.0 release), the database needs to be updated too either rerunning the script or with an *SQL ALTER clause* to rewrite the one query to the table (well, one could add a check on the code too and rewrite the query there, but I think this isn't the way to go). Otherwise the code uses the new binaries, which try to read the newly added *Generation*, which isn't there as it's not in the query.

A few ways to avoid this would be to define the queries as ``SELECT *``, that is, read all, but that would increase I/O. The traffic here is light, I guess, but as a matter of a principle it is not good. The other one would be to define the queries as hardcoded string in the code, in which case one can alter them freely and the queries would be distributed with the binaries. As there's only a limited set of queries, this arrangement is much more beneficial.

2) As an added note, the order of the columns do not matter, just the names (and types).

3) This would get caught in a test, but I don't remember there were tests stressing this part of the system, the client gateway. Let's add a mental note if someone has spare time. I'm currently with a limited access to the computer (about 12 for the least).